packages/mobile/src/store/products/selectors.ts

```diff
@@ -0,0 +1,12 @@
+import { RootState } from '~/common/types/types';
+import { createSelector } from '@reduxjs/toolkit';
+
+const selectProductById = (id: string) =>
+  createSelector(
+    (state: RootState) => state.products.products,
+    (products) => {
+      return products.find((item) => item.id === id);
+    },
+  );
```

Comment:
Please refer to this link and rewrite accordingly: https://github.com/reduxjs/reselect#q-how-do-i-create-a-selector-that-takes-an-argument